R/preview-standalone.R

```diff
@@ -436,3 +436,83 @@ st_image_show <- function(path,
   )
   return(ans)
 }
+
+#' Render and save a table in png or pdf format
+#'
+#' @param x an stable object.
+#' @param file the final output file name; if `file` has `.tex` extension, the
+#' extension will be replaced with the value in `format`.
+#' @param stem used to form the final output file name; this argument will
+#' override what is provided in `file`.
+#' @param format the output file format.
+#' @param dir the final output directory.
+#' @param build_dir the directory where the image will be built.
+#' @param ... passed to [st_aspdf()] or [st_aspng()], depending on the value of
+#' `format`.
+#'
+#' @details
+#' The table will be built / rendered in `tempdir()`.
+#'
+#' @examples
+#' tab <- stable(stdata(), output_file = "foo.tex")
+#'
+#' \dontrun{
+#'   ans <- stable_save_image(tab format = "pdf")
+#'   basename(ans)
+#'   st_image_show(ans)
+#' }
+#'
+#' @seealso [st_as_image()]
+#'
+#' @export
+stable_save_image <- function(x,
+                              file = attr(x, "stable_file"),
+                              stem = NULL,
+                              format = c("png", "pdf"),
+                              dir = getOption("pmtables.dir"),
+                              build_dir = tempdir(), ...) {
```

Comment:
Plain `tempdir()` calls that give the top-level per-session temporary directory, instead of a within-session unique directory like you'd get with `tempfile()~/~withr::local_tempdir()`, make me a bit nervous, but it looks like this matches the behavior of `st_aspdf()` and `st_aspng()`, so it seems it's not a problem in this context and should be fine here.